Makefile

```diff
@@ -22,7 +22,8 @@ export KUBEBUILDER_CONTROLPLANE_START_TIMEOUT ?=60s
 export KUBEBUILDER_CONTROLPLANE_STOP_TIMEOUT ?=60s
 
 # Image URL to use all building/pushing image targets
-IMG ?= gcr.io/k8s-cluster-api/cluster-api-controller:latest
+CONTROLLER_IMG ?= gcr.io/k8s-cluster-api/cluster-api-controller:latest
+PROVIDER_IMG ?= gcr.io/k8s-cluster-api/example-provider-controller:latest
```

Comment:
We'll have to coordinate with @roberthbailey to get this image pushed and work out a strategy for updating as needed.

Reply:
Should this variable be prefixed with `EXAMPLE_`?

Reply:
@vincepri I understand these variables are part of the top-level `Makefile` and could be overridden for using a different provider. It should be the ci scrips which set the proper value for the provider in the context of the tests.

Reply:
Yeah, that make sense. I was hoping we could prefix `PROVIDER_IMG` with `EXAMPLE_` to indicate this is the example provider.

Reply:
i will add the example prefix, thanks

Reply:
@detiber this PR effectively does not change the way we publish official cluster-api container (kept the same repo name), hence i will defer the discussion with @roberthbailey later, i do agree we need to follow up on controller container publishing.